spider_bot.py

```diff
@@ -1,46 +1,59 @@
-from irc import *
-from secrets import *
-import os
-import random
-import time
+from irc import IRC
+from secrets import get_server, get_channel
+from time import sleep
+
 
 channel = get_channel()
 server = get_server()
 nickname = "spider_bot"
 
+
 irc = IRC()
 irc.connect(server, channel, nickname)
 
 
-while 1:
-    text = irc.get_text()
-    print text
+def check_msg(text):
+    return "PRIVMSG" in text and channel in text
+
+
+def spider_bot(irc, nickname):
+    # Infinite loop until disconnected
+    while True:
+        text = irc.get_text()
+        print text
+
+        if check_msg(text) and "*snap*" in text.lower():
+            irc.send(channel, "Alexis...I don't feel so good...")
+            irc.quit("disintegrated")
+
+            if nickname == "spider_bot":
+                nickname = "CAPTAIN_MARVEL"
+            elif nickname == "CAPTAIN_MARVEL":
+                nickname = "spider_bot"
 
-    if "PRIVMSG" in text and channel in text and "*snap*" in text.lower():
-        irc.send(channel, "Alexis...I don't feel so good...")
-        irc.quit("disintegrated")
+            sleep(5)
 
-        if nickname == "spider_bot":
-            nickname = "CAPTAIN_MARVEL"
-        elif nickname == "CAPTAIN_MARVEL":
-            nickname = "spider_bot"
+            irc = IRC()
+            irc.connect(server, channel, nickname)
 
-        time.sleep(5)
+        if check_msg(text) and "goat" in text.lower():
+            irc.send(channel, "THE HERD")
 
-        irc = IRC()
-        irc.connect(server, channel, nickname)
+        if check_msg(text)\
+           and ("tom brady" in text.lower()
+                or "tb12" in text.lower()
+                or "touchdown tommy" in
+                text.lower()):
+            irc.send(channel, "the GOAT")
 
-    if "PRIVMSG" in text and channel in text and "goat" in text.lower():
-        irc.send(channel, "THE HERD")
+        if check_msg(text) and "spider_bot" in text.lower():
+            irc.send(channel, ":D")
 
-    if "PRIVMSG" in text and channel in text and ("tom brady" in text.lower() or "tb12" in text.lower() or "touchdown tommy" in text.lower()):
-        irc.send(channel, "the GOAT")
+        if check_msg(text) and "the herd" in text.lower():
+            irc.send(channel, "ALL THE GOATS")
 
-    if "PRIVMSG" in text and channel in text and "spider_bot" in text.lower():
-        irc.send(channel, ":D")
+        if check_msg(text) and "idea" in text.lower():
+            irc.send(channel, "it's gonna be HUUUGGEEE")
 
-    if "PRIVMSG" in text and channel in text and "the herd" in text.lower():
-        irc.send(channel, "ALL THE GOATS")
 
-    if "PRIVMSG" in text and channel in text and "idea" in text.lower():
-        irc.send(channel, "it's gonna be HUUUGGEEE")
+spider_bot(irc, nickname)
```

Comment:
Any reason spider_bot doesn't have the `if __name__ == "__main___"` part here? Also, I don't think the irc instance needs to be passed in. It isn't passed in for thanos or emoji_bot.

Reply:
re if __name__: I had accommodated your pastebin changes before your 180 turn. Forgot to change it back to my version.

re irc instance: In thanos_bot.py and emoji_bot.py, we don't use "irc.send()" and "irc.quit()" as we do in this file. Try testing it without passing in the instance as an argument and you'll get a compilation error.